### PR TITLE
CRUD for Comment

### PR DESCRIPTION
**Pull request by lkrnetic**
Created CRUD for Comment (including validation). Edited Post so that it can use mapper for converting from DTO to Post. Added data for initialization.

### CODE REVIEW

File: src/main/java/com/example/SimplifiedReddit/service/impl/SubredditServiceImpl.java [hunk 1]
```diff
@@ -48,9 +47,9 @@ public Subreddit createSubreddit(SubredditDTO subredditDTO) throws ConflictExcep
         return subredditRepository.save(subreddit);
     }
 
+    @Transactional(readOnly = true)
```
Comment (gobonja) [inline src/main/java/com/example/SimplifiedReddit/service/impl/SubredditServiceImpl.java]: zašto je delete readonly?

File: src/main/java/com/example/SimplifiedReddit/service/impl/CommentServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        if (!Objects.equals(comment.getUser().getId(), commentDTO.getUserId())) {
+            throw new ConflictException("Given id of comment doesn't belong to user who created comment with given id.");
+        }
+
+        commentDTO.setId(id);
+
+        return commentRepository.save(commentMapper.commentDTOtoComment(commentDTO));
+    }
+
+    @Transactional(readOnly = true)
```
Comment (gobonja) [inline src/main/java/com/example/SimplifiedReddit/service/impl/CommentServiceImpl.java]: zašto je delete readonly?

File: src/main/java/com/example/SimplifiedReddit/controller/CommentController.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    public CommentController(CommentService commentService, CommentMapper commentMapper) {
+        this.commentService = commentService;
+        this.commentMapper = commentMapper;
+    }
+
+    @GetMapping(params = {"userId"})
+    public ResponseEntity<?> getCommentsByUserId(@RequestParam Long userId) {
+        System.out.println("blabla1");
+        System.out.println(userId);
```
Comment (gobonja) [inline src/main/java/com/example/SimplifiedReddit/controller/CommentController.java]: što ovo radi u PR-u?

File: src/main/java/com/example/SimplifiedReddit/controller/CommentController.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            headers.add(ERROR_MESSAGE_KEY, exception.getMessage());
+            return new ResponseEntity<>(headers, HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @DeleteMapping(params = {"id"})
+    public ResponseEntity<?> deleteComment(@RequestParam  Long id) {
+        try {
+            commentService.deleteComment(id);
+            return new ResponseEntity<>(HttpStatus.OK);
```
Comment (gobonja) [inline src/main/java/com/example/SimplifiedReddit/controller/CommentController.java:83]: kod uspješnog delete-a vrati 204 No Content

**Review by gobonja — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/SimplifiedReddit/controller/CommentController.java (hunk 3 above)
Comment (lkrnetic) [reply src/main/java/com/example/SimplifiedReddit/controller/CommentController.java]: da kasnije sam skonto

**Pull request merged**